nix/build.sh

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env bash
+
+set -e
+
+# cleanup for artefacts created during builfs
+function cleanup() {
+  trap - EXIT ERR INT QUIT
+
+  if [ -n "$nixResultPath" ]; then
+    echo "Deleting derivations from Nix store..."
+    releaseDrv=$(nix-instantiate --quiet $nixOpts)
+    if [ -n "$releaseDrv" ]; then
+      local releaseSrcPath=$(nix-store -q --binding src $releaseDrv)
+      local releaseOutPath=$(nix-store -q --outputs $releaseDrv)
+      local releaseRefs=( $(nix-store -q --references $releaseDrv) )
+      local prodBuildDrv=$(printf -- '%s\n' "${releaseRefs[@]}" | grep -e "prod-build-android.drv")
```

Comment:
I guess in this area we'll just blindly look for references we don't want to keep (such as prod-build-{android,ios,desktop}), and ignore if none are present.